### ISSUE TITLE
A way to request the current scope

### ISSUE THREAD

**Issue by swissonid** [fluttercommunity/get_it#153]
Let's say you want to be sure that each time you enter a specific section of your app, that you start a new clean session. 
So before you start a new one you wanna check the current scope, to know if it is nesscacy to pop.

Right now something like that:

```dart
Future<void> startSession() async {
    final sessionScopeName = 'sesstionScopeName';
    if(getIt.currentScopeName == sessionScopeName) { <-- New Feature 
      await geit.popScope();// to be sure that the old session gets removed
    }
    await pushNewScope(scopeName: sessionScopeName);
}
```

is not possible and because of the issue #152 you can not relay on any exception. The only way I found was to add ah dummy scope which can be pop without any issue.

So it would be nice to have the option to request the current scope. Or do I overlook something?

**Comment by escamoteur** [comment]
I thinks that's an absolute reasonable request. But probably it will be added to the null save version.
In general I would feel quite uncomfortable when I had to check if my app is in a certain state.
If you have to I'm not sure if your app logic is really save.

**Comment by xmkevinchen** [comment]
Really looking forward to seeing this enhancement would be in the `get_it` upcoming release.

Right now, I need to manually check some instances registered or not to determine whether the scope has been pushed or not.

**Comment by escamoteur** [comment]
Uh, I somehow forgot about this. will add it soon

**Comment by escamoteur** [comment]
Implemented in V6.1.0
